tools/discover_gtests.sh

```diff
@@ -78,7 +91,14 @@ then
     testname=$(basename $test_binary)
 fi
 
-tests=$($test_binary --gtest_list_tests --gtest_filter=-$excludes | grep -v '^ ' | cut -d' ' -f1 | grep '\.' | sed 's/$/*/')
+if [ -z "$includes" ];
+then
+    discover_filter="-$excludes"
+else
+    discover_filter="$includes:-$excludes"
+fi
```

Comment:
If you decide against "avoiding" colons where I commented above, then this would be in the same vein?

```suggestion
discover_filter="$includes:-$excludes"
```

Reply:
Can't we have the best of both?

```suggestion
    discover_filter="${includes:+$includes:}-$excludes"
```

Reply:
We can!